submit-api/src/submit_api/models/account.py

```diff
@@ -16,4 +16,22 @@ class Account(BaseModel):
     __tablename__ = 'accounts'
 
     id = Column(db.Integer, primary_key=True, autoincrement=True)
-    proponent_id = Column(db.String())
+    proponent_id = Column(db.String(), nullable=False, unique=True)
+
+    @classmethod
+    def get_by_proponent_id(cls, proponent_id) -> Account:
+        """Fetch account by proponent id."""
+        return cls.query.filter_by(proponent_id=proponent_id).first()
+
+    @classmethod
+    def create_account(cls, account_data, session=None) -> Account:
+        """Create account."""
+        account = Account(
+            proponent_id=account_data.get('proponent_id', None),
+        )
+        if session:
```

Comment:
May I know the purpose of this checking "if session"

Reply:
if we have 3 transactions and one fail and we need to rollback everything if one fails I implemented the above pattern.

you pass the same session for all 3. if no session is passed then it's just db.session.commit or something

Reply:
![image](https://github.com/user-attachments/assets/94911008-38d2-41c0-a252-4ad77a40c906)

Reply:
I got it. I hope we will be using it only  in those scenarios where there are multiple database transactions involved.